transform.ml

```diff
@@ -0,0 +1,89 @@
+open Instr
+
+let remove_empty_jmp prog =
+  let rec remove_empty_jmp pc =
+    let pc' = pc + 1 in
+    if pc' = Array.length prog then [prog.(pc)] else
+      match (prog.(pc), prog.(pc')) with
+      | (Goto l1, Label l2) when l1 = l2 ->
```

Comment:
is that correct if other instructions can jump to `l2`? I'd think you need to keep `Label l2` in the output unless you can check that it has a single predecessor.